src/main/js/component/general/distribution/JobAddModal.js

```diff
@@ -24,23 +25,30 @@ class JobAddModal extends Component {
         this.createJobTypeOptions = this.createJobTypeOptions.bind(this);
     }
 
+    componentDidMount() {
+        this.props.resetDistributionDescriptor();
+    }
+
     getCurrentJobConfig() {
         switch (this.state.values.typeValue) {
             case 'channel_email':
                 return (<GroupEmailJobConfiguration
+                    alertChannelName={this.state.values.typeValue}
```

Comment:
These all have the same alertChannelName?

Reply:
The property has to be named alertChannelName because the slack configuration that is read has a property channel_name.  The name of this property needed to be different in order to pass the correct channel name.  The this.state.values.typeValue is "channel_email", "channel_hipchat", or "channel_slack" based on what the user selected from the drop down. We want that value in alertChannelName in order to send a get request to get the right UI components for the distribution job.  In the future when we have a React component that renders the configuration screen based on the ui components from the server we won't need this switch statement.  We will just pass in the value for this.state.values.typeValue.